### PR TITLE
fix: Close icon style problem when the TextArea attribute resize is n…

### LINKED ISSUE #39099: [Input.TextArea]: resize 不为 none 时 close 和 feedback 图标样式问题
### Reproduction link

[https://ant.design/~demos/components-form-demo-validate-static](https://ant.design/~demos/components-form-demo-validate-static)

### Steps to reproduce

调整文本域宽度即可复现

### What is expected?

close 和 feedback 图标在文本域内

### What is actually happening?

close 和 feedback 图标在文本域外

| Environment | Info |
| --- | --- |
| antd | 5.0.2 |
| React | 18.2.0 |
| System | MACOS 12.1 |
| Browser | chrome 107.0.5304.121 |

---
4.x 版本的文本域仅支持高度调整所以没有这个问题，5.x 版本同时支持了高度和宽度的调整

image

### PR DESCRIPTION
**Pull request by wangping51**
…ot none

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

fix #39099 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   fix Close icon style problem when the TextArea attribute resize is not none   |
| 🇨🇳 Chinese |   修复了当TextArea属性的resize不是none时关闭图标样式的问题       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/input/style/index.tsx [hunk 1]
```diff
@@ -592,6 +592,7 @@ const genAffixStyle: GenerateStyle<InputToken> = (token: InputToken) => {
     [`${componentCls}-affix-wrapper`]: {
       ...genBasicInputStyle(token),
       display: 'inline-flex',
+      width: 'auto',
```
Comment (afc163) [inline components/input/style/index.tsx:595]: 貌似并不能解决问题。

File: components/input/style/index.tsx [hunk 2]
```diff
@@ -566,6 +566,7 @@ const genAllowClearStyle = (token: InputToken): CSSObject => {
     '&-textarea-with-clear-btn': {
       padding: '0 !important',
       border: '0 !important',
+      width: 'auto',
```
Comment (afc163) [inline components/input/style/index.tsx:569]: 这个解决不了问题

File: components/input/style/index.tsx (hunk 1 above)
Comment (51wangping) [reply components/input/style/index.tsx]: 已修改方案。
并发现问题: TextArea showCount 时 data-count 样式不跟随文本域大小调整

File: components/input/style/index.tsx (hunk 2 above)
Comment (51wangping) [reply components/input/style/index.tsx:569]: 额， 我这边看文档中的demo，图标已经能跟随 textarea 大小变化了
image

Comment (51wangping) [reply components/input/style/index.tsx:569]: 同时发现 textarea 属性为 showCount 时， data-count 不会跟随 textarea 大小变化

Comment (afc163) [reply components/input/style/index.tsx:569]: feedback 貌似不行

Comment (afc163) [reply components/input/style/index.tsx:569]: 4.x 有默认 reset 样式，5.x 没了，单独加到组件里。https://github.com/ant-design/ant-design/pull/39118

**Pull request closed without merging**